app/controllers/search_controller.rb

```diff
@@ -11,4 +11,14 @@ def extract_filters
     respond_with TextSearchClient.parse_text(params[:q])
   end
 
+  def log_metrics_event
+    if params[:type] && params[:data] then
+      if params[:other_data] && params[:other_data] != "" then
+        metrics_event(params[:type], params[:data], eval(params[:other_data])) 
```

Comment:
why ```eval```?

Reply:
It's sent in as a string; however the function in application_controller defines it as an hash.  Eval converts it so that the parameter is acceptable by metrics_event().

Reply:
```eval(params[:other_data])``` => ```{other_data: params[:other_data]}```

Reply:
That didn't work as well.  Metrics output using that syntax:
[metrics] {"event":"reverb_redirect","data":"back_to_reverb","session":"780241f7be3da5cad2f823b7a094c805","timestamp":1506531924415,"{:source=\u003e\"modal link\"}":{"other_data":"{source: 'modal link'}"}}

Note the "\u003e\" stuff.

The line I used doesn't have that and is easier to read.  It works - why fix it?

Reply:
I am not asking you to fix it. See my overall comment. We currently only need to address 1. security issues and 2. the ENV values which will fix some of the issues in 1.